src/components/browse/singleDatasetView/singleDatasetViewAccordion/singleDatasetViewAccordion.tsx

```diff
@@ -33,7 +35,19 @@ const SingleDatasetViewAccordion = (props: SingleDatasetViewAccordionProps) => {
       const files = (props.details as any)[key];
       const file_category =
         key.charAt(0).toUpperCase() + key.slice(1, -1).replaceAll("_", " ");
-      return files.map((file: any) => ({ ...file, file_category }));
+      const files_with_info = files.map((x: any) => {
+        const found_file = props.files.file_information.find(
```

Comment:
I don't understand what you mean by this. Which array convert to map? props.files? I don't understand how this would help.

Reply:
See 99af162. The idea is to first convert the `file_information` array into a map, so that we only need to do a lookup inside the map method instead of a find operation which is much slower.